File: src/main/java/com/ownwn/config/ModMenu.java
package com.ownwn.config;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

public class ModMenu implements ModMenuApi {
    // this class exists so that you can open the gui from the ModMenu mod. look at the "Modmenu" entrypoint in fabric.mod.json
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return ConfigScreen::new;
    }
}

File: src/main/kotlin/com/ownwn/config/ModMenu.kt
package com.ownwn.config

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment

@Environment(EnvType.CLIENT)
class ModMenu : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory { parent ->
            NewConfig.createScreen(parent)
        }
    }
}